library/alloc/src/collections/vec_deque/iter.rs

```diff
@@ -94,18 +100,18 @@ impl<'a, T> Iterator for Iter<'a, T> {
         let (mut iter, final_res);
         if self.tail <= self.head {
             // Safety: single slice self.ring[self.tail..self.head] is initialized.
-            iter = unsafe { MaybeUninit::slice_assume_init_ref(&self.ring[self.tail..self.head]) }
-                .iter();
+            let sli: &[T] = unsafe { &(&*self.ring)[self.tail..self.head] };
```

Comment:
I suspect this reborrowing pattern makes people uncomfortable, but this passes Miri, and I'm not really sure how to avoid constructing `&[T]`. Perhaps in this case it is worth the extra code to only construct the slice to the init region?